src/app/users/get-user.ts

```diff
@@ -1,65 +1,99 @@
-import {
-  SemanticAttributes,
-  asyncRunInSpan,
-  addAttributesToCurrentSpan,
-  ENDUSER_ALIAS,
-} from "@services/tracing"
+import { SemanticAttributes, asyncRunInSpan } from "@services/tracing"
 import { UsersRepository } from "@services/mongoose"
-import { IpFetcher } from "@services/ipfetcher"
 import { getIpConfig } from "@config/app"
+import { IpFetcher } from "@services/ipfetcher"
+import { UsersIpRepository } from "@services/mongoose/users-ip"
 import { RepositoryError } from "@domain/errors"
 
 const users = UsersRepository()
+const usersIp = UsersIpRepository()
 
 export const getUserForLogin = async ({
   userId,
   ip,
+  logger,
 }: {
-  userId: string
-  ip?: string
+  userId: UserId
+  ip?: Ip
+  logger: Logger
 }): Promise<User | ApplicationError> =>
   asyncRunInSpan(
     "app.getUserForLogin",
     { [SemanticAttributes.CODE_FUNCTION]: "getUserForLogin" },
     async () => {
-      const user = await users.findById(userId as UserId)
+      const user = await users.findById(userId)
+
       if (user instanceof Error) {
         return user
       }
-      addAttributesToCurrentSpan({
-        [ENDUSER_ALIAS]: user.username,
-      })
-      user.lastConnection = new Date()
 
-      // IP tracking logic could be extracted into domain
+      // this routing run asynchrously, to update metadata on the background
+      updateUserIpInfo({ userId, ip, logger })
+
+      return user
+    },
+  )
+
+const updateUserIpInfo = async ({
+  userId,
+  ip,
+  logger,
+}: {
+  userId: UserId
+  ip?: Ip
+  logger: Logger
+}): Promise<void> =>
+  asyncRunInSpan(
+    "app.addIp",
```

Comment:
This should be the same as the name of the function `app.updateUserIpInfo`

Reply:
some DRY would be great here. not sure if that could work [Function.name](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Function/name) as it's called within the function

Reply:
Agree - I think @dolcalmi wanted to think about a good way to add tracing to app + service layer via wrapping our index files anyway. Until he's spent some effort here I wouldn't change anything.